slack/decorators/workflow.py

```diff
@@ -0,0 +1,59 @@
+import logging
+logger = logging.getLogger(name="statuspage init")
+
+from django.db.models.signals import post_save, post_migrate
+from django.db.utils import ProgrammingError
+from django.dispatch import receiver
+
+from slack.apps import SlackConfig
+from slack.models import Workflow, WorkflowParameters, CommsChannel, workflow_post_save
+
+RESPONSE_WORKFLOWS = {}
+
+
+def register_workflow(name, *args, **kwargs):
+    def _wrapper(wf):
+        imp = wf(*args, **kwargs)
+        RESPONSE_WORKFLOWS[name] = imp
+        try:
+            wf, created = Workflow.objects.get_or_create(name=name)
+            if wf.enabled:
+                imp.enabled(wf)
+            else:
+                imp.disabled(wf)
+            return wf
+        except ProgrammingError as ex: # Throws errors before migration can be completed (first run)
+            logger.error(ex)
+    return _wrapper
+
+
+@receiver(post_migrate, dispatch_uid='setup_workflows') # Ensure everything is setup after migration
```

Comment:
Is it necessary for this function to run after every migration we run in the Slack app?  It looks like this is just required after the migration to add workflows, in which case it could be add to the migrations file, as a `RunPython` command at the end of the `operations` list (I think).

Reply:
Was able to just remove it after some changes to the decorator.